legend-pure-runtime/legend-pure-runtime-java-engine-compiled/src/main/java/org/finos/legend/pure/runtime/java/compiled/generation/processors/support/Pure.java

```diff
@@ -736,9 +742,38 @@ public void safeValue(KeyExpression o) throws Exception
                     {
                         m.invoke(result, Lists.fixedSize.of(res));
                     }
+
+                }
+            });
+
+            RichIterable<CoreInstance> classProperties = _Class.getSimpleProperties(aClass, ((CompiledExecutionSupport) es).getProcessorSupport());
+            classProperties.forEach(new CheckedProcedure<CoreInstance>()
+            {
+                @Override
+                public void safeValue(CoreInstance p) throws Exception
+                {
+                    if (!userAssignedParameters.contains(p.getName()))
+                    {
+                        Property<?, ?> prop = (Property<?, ?>) p;
+                        DefaultValue defaultValue = prop._defaultValue();
+                        if (defaultValue != null)
+                        {
+                            Object res = reactivate(defaultValue._functionDefinition()._expressionSequence().getFirst(), new PureMap(Maps.fixedSize.empty()), bridge, es);
+                            Method method = c.getMethod("_" + prop._name(), RichIterable.class);
+                            if (res instanceof RichIterable)
+                            {
+                                method.invoke(result, res);
+                            }
+                            else
+                            {
+                                method.invoke(result, Lists.fixedSize.of(res));
+                            }
+                        }
+                    }
```

Comment:
We have methods now on AbstractCompiledCoreInstance for dealing with default values. Maybe result above can be cast directly to AbstractCompiledCoreInstance? In case it needs to be cast to Any above, I'm casting it in the suggestion below.
```suggestion
            AbstractCompiledCoreInstance instance = (AbstractCompiledCoreInstance) result;
            instance.getDefaultValueKeys().forEach(new CheckedProcedure<String>()
            {
                @Override
                public void safeValue(String p) throws Exception
                {
                    if (!userAssignedParameters.contains(p))
                    {
                        RichIterable<?> defaultValues = instance.getDefaultValue(p, es);
                        if ((defaultValues != null) && defaultValues.notEmpty())
                        {
                            Method method = c.getMethod("_" + prop._name(), RichIterable.class);
                            method.invoke(result, defaultValues);
                        }
                    }
```